cmd/tsbs_generate_data/common/distribution_test.go

```diff
@@ -0,0 +1,70 @@
+package common
+
+import (
+	"math"
+	"testing"
+)
+
+const (
+	tolerance = 0.0000000001
+)
+
+type mockDistribution struct {
+	AdvanceCalled bool
+	ReturnValue   float64
+}
+
+func (m *mockDistribution) Advance() {
+	m.AdvanceCalled = true
+}
+
+func (m *mockDistribution) Get() float64 {
+	return m.ReturnValue
+}
+
+func TestFloatPrecisionAdvance(t *testing.T) {
+
+	dist := &mockDistribution{}
+
+	fp := FP(dist, 1)
+
+	fp.Advance()
+
+	if !dist.AdvanceCalled {
+		t.Errorf("FloatPrecision Advance call did not call underlying distribution Advance method")
+	}
+}
+
+func TestFloatPrecisionGet(t *testing.T) {
+	testCases := []struct {
+		value   float64
+		results map[int]float64
+	}{
+		{
+			value: 1.234567890,
```

Comment:
can you add another case for an integer like 1.0? and just for fun, 0.0 as well? those values seem "special" enough to have their own tests